### ISSUE TITLE
AssertionError where root is a const, not a module

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/astroid#2618]
The assertion added in 275f50841 fails when testing the provided file in pylint-dev/pylint#10024:

### Bug description

When parsing the following ``a.py``:

```python
"""Test module"""
from prefect import flow, task
from prefect.states import Completed

@task
def task1(name: str) -> str:
    return f"Your name is {name}."

@flow
def myflow():
    result = task1("Jim", return_state=True)
    return Completed(message=result.result())

```

### Command used

```shell
pylint a.py
```

### Pylint output

<details open>
    <summary>
        pylint crashed with a ``AstroidError`` and with the following stacktrace:
    </summary>

```python
Traceback (most recent call last):
  File "/Users/.../pylint/pylint/lint/pylinter.py", line 788, in _lint_file
    check_astroid_module(module)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/Users/.../pylint/pylint/lint/pylinter.py", line 1017, in check_astroid_module
    retval = self._check_astroid_module(
        ast_node, walker, rawcheckers, tokencheckers
    )
  File "/Users/.../pylint/pylint/lint/pylinter.py", line 1069, in _check_astroid_module
    walker.walk(node)
    ~~~~~~~~~~~^^^^^^
  File "/Users/.../pylint/pylint/utils/ast_walker.py", line 90, in walk
    self.walk(child)
    ~~~~~~~~~^^^^^^^
  File "/Users/.../pylint/pylint/utils/ast_walker.py", line 90, in walk
    self.walk(child)
    ~~~~~~~~~^^^^^^^
  File "/Users/.../pylint/pylint/utils/ast_walker.py", line 90, in walk
    self.walk(child)
    ~~~~~~~~~^^^^^^^
  [Previous line repeated 3 more times]
  File "/Users/.../pylint/pylint/utils/ast_walker.py", line 87, in walk
    callback(astroid)
    ~~~~~~~~^^^^^^^^^
  File "/Users/.../pylint/pylint/checkers/typecheck.py", line 1111, in visit_attribute
    if _is_owner_ignored(
       ~~~~~~~~~~~~~~~~~^
        owner,
        ^^^^^^
    ...<2 lines>...
        self.linter.config.ignored_modules,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/Users/.../pylint/pylint/checkers/typecheck.py", line 123, in _is_owner_ignored
    if is_module_ignored(owner.root().qname(), ignored_modules):
                         ~~~~~~~~~~^^
  File "/Users/.../astroid/astroid/nodes/node_ng.py", line 334, in root
    assert isinstance(self, nodes.Module)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
AssertionError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/.../pylint/pylint/lint/pylinter.py", line 752, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../pylint/pylint/lint/pylinter.py", line 790, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError

```

</details>

### Expected behavior

No crash.

### Pylint version

```shell
pylint 4.0.0-dev0
astroid 4.0.0-dev0
Python 3.13.0rc1 (v3.13.0rc1:e4a3e786a5e, Jul 31 2024, 19:49:53) [Clang 15.0.0 (clang-1500.3.9.4)]
```

***

cc/ @temyurchenko

**Comment by temyurchenko** [comment]
Hmm, I can't reproduce on the latest pylint main + astroid main, @jacobtylerwalls. I'm using python 3.12 though.

**Comment by jacobtylerwalls** [comment]
Sorry! I hadn't pulled the latest from main. Fixed in 7710b7b446e7d99fb0d6150707964d0e8d0d3c11.
